pep-0394.txt

```diff
@@ -5,104 +5,196 @@ Last-Modified: $Date$
+  The commands with no version number should either invoke the same version
+  of Python as the ``python`` command, or not be available at all.
+* When packaging third party Python scripts, distributors are encouraged to
+  change less specific shebangs to more specific ones.
+  This ensures software is used with the latest version of Python available,
+  and it can remove a dependency on Python 2.
+  The details on what specifics to set are left to the distributors;
+  though. Example specifics could include:
+
+  * Changing ``python`` shebangs to ``python3`` when Python 3.x is supported.
+  * Changing ``python`` shebangs to ``python2`` when Python 3.x is not yet
+    supported.
+  * Changing ``python3`` shebangs to ``python3.8`` if the software is built
+    with Python 3.8.
+
+* When a virtual environment (created by the PEP 405 ``venv`` package or a
+  similar tool such as ``virtualenv`` or ``conda``) is active, the ``python``
+  command should refer to the virtual environment's interpreter and should
+  always be available.
+  The ``python3`` or ``python2`` command (according to the environment's
+  interpreter version) should also be available.
+
+For Python script publishers
+----------------------------
+
 * When reinvoking the interpreter from a Python script, querying
   ``sys.executable`` to avoid hardcoded assumptions regarding the
   interpreter location remains the preferred approach.
-* In controlled environments aimed at expert users, where being explicit
-  is valued over user experience (for example, in test environments and
-  package build systems), distributions may choose to not provide the
-  ``python`` command even if ``python2`` is available.
-  (All software in such a controlled environment must use ``python3`` or
-  ``python2`` rather than ``python``, which means scripts that deliberately
-  use ``python`` need to be modified for such environments.)
-* When a virtual environment (created by the PEP 405 ``venv`` package or a
-  similar tool) is active, the ``python`` command should refer to the
-  virtual environment's interpreter. In other words, activating a virtual
-  environment counts as deliberate user action to change the default
-  ``python`` interpreter.
+* Encourage your end users to use a virtual environment.
+  This makes the user's environment more predictable (possibly resulting
+  in fewer issues), and helps avoid disrupting their system.
+* For scripts that are only expected to be run in an activated virtual
+  environment, shebang lines can be written as ``#!/usr/bin/env python``,
+  as this instructs the script to respect the active virtual environment.
+* In cases where the script is expected to be executed outside virtual
+  environments, developers will need to be aware of the following
+  discrepancies across platforms and installation methods:
+
+  * Older Linux distributions will provide a ``python`` command that
+    refers to Python 2. Most of these distros do *not* provide a
```

Comment:
The second sentence feels a little too imprecise.  What exactly is an "older Linux distribution" and how many of them count as "most"?  Perhaps:

> Older Linux distributions will provide a ``python`` command that refers to Python 2, and will likely not provide a ``python2`` command